tests/main/views/test_audits.py

```diff
@@ -887,6 +887,18 @@ def test_paginated_audit_with_invalid_custom_page_size(self):
         response = self.client.get('/audit-events?per_page=foo')
         assert response.status_code == 400
 
+    def test_should_get_audit_events_by_supplier_id_field_in_data(self):
+        for id, info in [(0, 'miss'), (3, 'hit'), (2, 'miss'), (7, 'miss'), (3, 'hit'), (1, 'miss')]:
+            self.add_audit_event(data={'supplierId': id, 'info': info})
+
+        response = self.client.get('/audit-events?data-supplier-id=3')
+        audit_events = json.loads(response.get_data())['auditEvents']
+
+        assert response.status_code == 200
+        assert len(audit_events) == 2
+        assert all(audit_event['data']['supplierId'] == 3 for audit_event in audit_events)
+        assert all(audit_event['data']['info'] == 'hit' for audit_event in audit_events)
```

Comment:
What happens when an AuditEvent doesn't have `supplierId` in `data`?

Reply:
Nothing (hopefully)! I've added a test to show this, but the index is set up to only index rows where supplier id is set.

Reply:
Nice 👍